website/addons/base/testing/views.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+import httplib as http
+import urlparse
+
+import mock
+from nose.tools import *  # noqa (PEP8 asserts)
+
+from framework.auth import Auth
+
+from website.addons.base.testing.base import OAuthAddonTestCaseMixin
+from website.util import api_url_for, web_url_for, permissions
+
+from tests.factories import AuthUserFactory, ProjectFactory
+
+
+class OAuthAddonAuthViewsTestCaseMixin(OAuthAddonTestCaseMixin):
+
```

Comment:
Should this class also have a test for deleting an external account with invalid credentials? Also, how about when oauth finish is canceled? In dropbox TestAuthViews, it was `test_dropbox_oauth_delete_user_with_invalid_credentials` and `test_dropbox_oauth_finish_cancelled` for reference.